lib/skate/settings/user.ex

```diff
@@ -75,12 +76,20 @@ defmodule Skate.Settings.User do
   end
 
   def is_in_test_group(user_id, test_group_name) do
-    %{test_groups: user_test_groups} =
-      DbUser
-      |> Skate.Repo.get(user_id)
-      |> Skate.Repo.preload(:test_groups)
+    user_id
+    |> get_by_id!()
+    |> all_test_group_names()
+    |> Enum.member?(test_group_name)
+  end
+
+  def all_test_group_names(user) do
+    user = Skate.Repo.preload(user, :test_groups)
+
+    enabled_test_groups = TestGroup.get_enabled()
 
-    Enum.any?(user_test_groups, &(&1.name == test_group_name))
+    (enabled_test_groups ++ user.test_groups)
```

Comment:
**suggestion:** I'm sort of torn because this logic (combine the user-specific test groups with the globally0enabled test groups) feels more like test group logic than user logic, but I can't quite think of another way to split it up.

Reply:
Count this piece of feedback as optional, though, as I'm not really sure what it should look like instead.

Reply:
Honestly, this felt like the sort of dumb hacky code that we'll refactor in the future. I don't like it either, but this is easy to fix after-the-fact.

Reply:
After a few minutes of thought, it feels like there are three separate things going on here:
 - Admin view / User management (seeing all users, for a given user, seeing what test groups they're in, etc)
 - Admin view / Test group management (a lot of the behavior in test_group.ex)
 - User view / Test group membership (when a user loads the regular skate app, what test groups are they a member of for purposes of `data-user-test-groups` and any other backend `is_in_test_group` calls)

I'm thinking that in the medium-term, each of these different questions belongs in its own module, with a new `TestGroupMembership` module (probably with a better name than that) that takes care of all of the increasingly complex logic for figuring out what test groups a user is a part of.

(Maybe that could also be an opportunity to move all of this test group stuff out of the `Settings` context, where it sort of feels out of place)